Extensions/keyboardShortcut.js

```diff
@@ -1,236 +1,102 @@
-	registerBind("H", false, true, false, clickNavigatingBackButton);
-	registerBind("L", false, true, false, clickNavigatingForwardButton);
-
-	// PageUp, PageDown to focus on iframe app before scrolling
-	registerBind("PAGE_UP", false, true, false, focusOnApp);
-	registerBind("PAGE_DOWN", false, true, false, focusOnApp);
-
-	// J and K to vertically scroll app
-	registerBind("J", false, false, false, appScrollDown);
-	registerBind("K", false, false, false, appScrollUp);
-
-	// G and Shift + G to scroll to top and to bottom
-	registerBind("G", false, false, false, appScrollTop);
-	registerBind("G", false, true, false, appScrollBottom);
-
-	// M to Like/Unlike track
-	registerBind("M", false, false, false, Spicetify.Player.toggleHeart);
-
-	// Forward Slash to open search page
-	registerBind("/", false, false, false, openSearchPage);
-
-	if (window.navigator.userAgent.indexOf("Win") === -1) {
-		// CTRL + Arrow Left Next and CTRL + Arrow Right  Previous Song
-		registerBind("ARROW_RIGHT", true, false, false, nextSong);
-		registerBind("ARROW_LEFT", true, false, false, previousSong);
-
-		// CTRL + Arrow Up Increase Volume CTRL + Arrow Down Decrease Volume
-		registerBind("ARROW_UP", true, false, false, increaseVolume);
-		registerBind("ARROW_DOWN", true, false, false, decreaseVolume);
-	}
-
-	// F to activate Link Follow function
-	const vim = new VimBind();
-	registerBind("F", false, false, false, vim.activate.bind(vim));
-	// Esc to cancel Link Follow
-	vim.setCancelKey("ESCAPE");
-
-	function rotateSidebarDown() {
-		rotateSidebar(1);
-	}
-
-	function rotateSidebarUp() {
-		rotateSidebar(-1);
-	}
-
-	function clickQueueButton() {
-		document.querySelector(".main-nowPlayingBar-right .control-button-wrapper > button").click();
-	}
-
-	function clickNavigatingBackButton() {
-		document.querySelector(".main-topBar-historyButtons .main-topBar-back").click();
-	}
-
-	function clickNavigatingForwardButton() {
-		document.querySelector(".main-topBar-historyButtons .main-topBar-forward").click();
-	}
-
-	function appScrollDown() {
-		const app = focusOnApp();
-		if (app) {
-			const scrollInterval = setInterval(() => {
-				app.scrollTop += SCROLL_STEP;
-			}, 10);
-			document.addEventListener("keyup", () => {
-				clearInterval(scrollInterval);
-			});
+	const binds = {
+		// Rotate through sidebar items using Ctrl+Tab and Ctrl+Shift+Tab
+		"ctrl+tab": { callback: () => rotateSidebar(1) },
+		"ctrl+shift+tab": { callback: () => rotateSidebar(-1) },
+
+		// Focus on the app content using Shift+PageUp and Shift+PageDown
+		"shift+pageup": { callback: focusOnApp },
+		"shift+pagedown": { callback: focusOnApp },
+
+		// Scroll actions using 'j' and 'k' keys
+		j: { callback: createScrollCallback(SCROLL_STEP) },
+		k: { callback: createScrollCallback(-SCROLL_STEP) },
+
+		// Scroll to the top ('g') or bottom ('Shift+g') of the page
+		g: { callback: scrollToPosition(0) },
+		"shift+g": { callback: scrollToPosition(1) },
```

Comment:
Uniformity with the previous diff, modify the functions related accordingly.
```suggestion
		j: { callback: () => createScrollCallback(SCROLL_STEP) },
		k: { callback: () => createScrollCallback(-SCROLL_STEP) },

		// Scroll to the top ('g') or bottom ('Shift+g') of the page
		g: { callback: () => scrollToPosition(0) },
		"shift+g": { callback: () => scrollToPosition(1) },
```

Reply:
have you tested this works since when I did this it did not.

Reply:
> have you tested this works since when I did this it did not.

> **modify the functions related accordingly**

The functions you wrote in question returns a **callback function** with the memorized values of the constructor function. Modify them to only execute actions instead of another function in combination with this suggestion.

Reply:
read the code this is already in place.

Reply:
![image](https://github.com/spicetify/spicetify-cli/assets/77577746/943becec-6b9f-401c-9c39-367a9eaad771)

Reply:
Refrain from closing suggestions when the discussion is not resolved.

Reply:
suggestions have been added